src/vectorpostprocessors/ResponseHistoryMean.C

```diff
@@ -0,0 +1,81 @@
+// MOOSE includes
+#include "ResponseHistoryMean.h"
+#include "PostprocessorInterface.h"
+#include "VectorPostprocessorInterface.h"
+#include "MastodonUtils.h"
+#include "ResponseHistoryBuilder.h"
+/* #include <iostream>
+#include <fstream>
+using namespace std;*/
+
+registerMooseObject("MastodonApp", ResponseHistoryMean);
+
+template <>
+InputParameters
+validParams<ResponseHistoryMean>()
+{
+  InputParameters params = validParams<GeneralVectorPostprocessor>();
+  params.addRequiredParam<VectorPostprocessorName>(
+      "vectorpostprocessor",
+      "Name of the ResponseHistoryBuilder vectorpostprocessor, for which "
+      "response spectra are calculated.");
+
+  // Make sure that csv files are created only at the final timestep
+  params.set<bool>("contains_complete_history") = true;
+  params.suppressParameter<bool>("contains_complete_history");
+
+  params.set<ExecFlagEnum>("execute_on") = {EXEC_FINAL};
+  params.suppressParameter<ExecFlagEnum>("execute_on");
+
+  params.addClassDescription("Calculate the mean acceleration time series given a set of nodes.");
+  return params;
+}
+
+ResponseHistoryMean::ResponseHistoryMean(const InputParameters & parameters)
+  : GeneralVectorPostprocessor(parameters),
+    _history_time(getVectorPostprocessorValue("vectorpostprocessor", "time"))
+
+{
+}
+
+void
+ResponseHistoryMean::initialSetup()
+{
+  const ResponseHistoryBuilder & history_vpp = getUserObjectByName<ResponseHistoryBuilder>(
```

Comment:
This also should be in the initialization list

```
_builder(getUserObject<ResponseHistoryBuilder>("vectorpostprocessor")),
```

Reply:
Transferred _builder to the initialization list.